### PR TITLE
Feature: Export Activities to Calendar

### PR DESCRIPTION
**Pull request by mohamedtahaguelzim**
#### Description
This PR introduces the `ExportToCalendarButton` composable, enabling users to export activity details to their phone's calendar. The button integrates with the system calendar using `Intent.ACTION_INSERT` and pre-fills the event details such as title, description, location, start time, and duration.

#### Key Changes
- Added `ExportToCalendarButton.kt`:
  - A Composable function that displays a button for exporting activities.
  - Leverages `Intent.ACTION_INSERT` to open the system's calendar app.
  - Handles optional location and duration fields.
- Includes a helper function `exportToCalendar` for creating and launching the calendar intent.
- Added a `PreviewExportButton` Composable.

#### How It Works
1. Users click the "Export to Calendar" button.
2. The app prepares the event data from the `Activity` model in the view model.
3. Launches the calendar app with the event details pre-filled.

#### Notes
- The duration is assumed to be in minutes and calculated relative to the start time.
- Future iterations can improve handling of time zones or add custom styling for the button.
- We need to consider the timezones in our code logic because it affects the calendar set up time functionnality. This was not fixed as it goes beyond the scope of this PR.

### CODE REVIEW

File: app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        exportToCalendar(
+            context = context,
+            activityTitle = activity.title,
+            description = activity.description,
+            location = activity.location?.name ?: "Unknown Location",
+            startMillis = startMillis,
+            endMillis = endMillis)
+      },
+      modifier = Modifier.fillMaxWidth().padding(16.dp)) {
```
Comment (hmaryam84) [inline app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt:43]: please, remove this magic constant.

File: app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+          putExtra(
+              CalendarContract.Events.EVENT_TIMEZONE,
+              "Europe/Zurich") // timezone is fixed, should be dynamic
+        }
+
+    // Check if an app exists to handle the intent
+    if (intent.resolveActivity(context.packageManager) != null) {
+      context.startActivity(intent)
+    } else {
+      Toast.makeText(context, "No apps available to handle calendar events", Toast.LENGTH_SHORT)
```
Comment (hmaryam84) [inline app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt:75]: Please put this string in strings.xml for a better modularization

File: app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+          .show()
+    }
+  } else {
+    Toast.makeText(
+            context,
+            "This activity is missing data. Unable to export to calendar.",
```
Comment (hmaryam84) [inline app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt:81]: Same comment as above

**Review by hmaryam84 — CHANGES_REQUESTED** [review]
The feature is very well implemented. There are some changes that must be made. Please refere to the comments I made. FurtherMore, there are no tests.  I am aware that this can be hard to test especially with the intents, but you can at least test the toasts and the presence of the button.

File: app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt (hunk 1 above)
Comment (mohamedtahaguelzim) [reply app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt]: Done.

File: app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt (hunk 2 above)
Comment (mohamedtahaguelzim) [reply app/src/main/java/com/android/sample/ui/activitydetails/ExportToCalendarButton.kt]: Done

**Review by hmaryam84 — APPROVED** [review]
Thank you a lot for having followed my suggestions. Even though the covergae is not very high, it not a big problem. It is indeed very hard to test such features.  The magic constants have been removed. Well done!

**Pull request merged**